gulpfile.js

```diff
@@ -1,29 +1,101 @@
 const forever = require('forever-monitor');
 const gulp = require('gulp');
+const gulpBabel = require('gulp-babel');
 const eslint = require('gulp-eslint');
 const mocha = require('gulp-mocha');
+const sourcemaps = require('gulp-sourcemaps');
+const webpackStream = require('webpack-stream');
+const webpackConfig = require('./webpack.config.js');
+
+/**
+ * @summary Delete the dist/ directory
+ */
+const babelClean = () => del(['dist']);
+
+/**
+ * @summary Copy files that don't need to be compiled from src/ to dist/
+ */
+const babelCopy = () => gulp.src(['src/**', '!src/**/*.js', '!src/tests/integration/**'])
+  .pipe(gulp.dest('dist'));
+
+/**
+ * @summary Compile JavaScript files and place them in dist/. Also, generate sourcemaps
+ */
+const babelCompile = () => gulp.src(['src/**/*.js'])
+  .pipe(sourcemaps.init())
+  .pipe(gulpBabel())
+  .pipe(gulp.dest('dist'));
+
+const babel = gulp.series(babelClean, babelCopy, babelCompile);
 
 /**
  * @summary Use Eslint linting *.js file besides source files in node_modules
  */
-gulp.task('lint', () => gulp.src(['**/*.js', '!node_modules/**'])
+const lint = () => gulp.src(['src/**/*.js', '!node_modules/**', '!dist/**'])
   .pipe(eslint())
   .pipe(eslint.format())
-  .pipe(eslint.failAfterError()));
+  .pipe(eslint.failAfterError());
+
+/**
+ * @summary Check Flow types
+ */
+const typecheck = () => spawn('./node_modules/.bin/flow', ['check'], { stdio: 'inherit' });
+
+/**
+ * @summary Run unit tests
+ */
+const test = () => gulp.src(['dist/tests/unit/*.js'])
+  .pipe(mocha({ reporter: 'spec' }));
 
 /**
- * @summary Run unit test
+ * @summary Bundle using webpack
  */
-gulp.task('test', () => gulp.src(['tests/unit/*.js'])
-  .pipe(mocha({ reporter: 'spec' })));
+const bundle = () => gulp.src('dist/app.js')
+  .pipe(webpackStream(webpackConfig))
+  .pipe(gulp.dest('dist/'));
 
 /**
  * @summary Start application using forever
  */
-gulp.task('start', () => new forever.Monitor('app.js').start());
+const start = () => new forever.Monitor('dist/bundle.js').start();
 
+/**
+ * @summary Lint and compile, test, and bundle the application
+ */
+const build  = gulp.series(gulp.parallel(lint, typecheck, babel), gulp.parallel(test, bundle));
 
 /**
- * @summary Run test and lint task parallelly before start the apllication
+ * @summary Start the application only
+ */
+exports.start = start;
+/**
+ * @summary Build the application only
+ */
+exports.build = build;
+/**
+ * @summary Builds and starts
+ */
+exports.devRun = gulp.series(build, start);
+/**
+ * @summary Compile and test the application only
+ */
+exports.test = gulp.series(babel, test);
+/**
+ * @summary Compile the code only
+ */
+exports.babel = babel;
+/**
+ * @summary Lint the code only
+ */
+exports.lint = lint;
+/**
+ * @summary Typecheck the code only
+ */
+exports.typecheck = typecheck;
+/**
+ * @summary Bundle the code only
  */
-gulp.task('run', gulp.series(gulp.parallel('lint', 'test'), 'start'));
+exports.bundle = bundle;
```

Comment:
Since we have quite a few tasks now, could we organize them a little? One way could be to have a few sections
* All function definitions for tasks that aren't just a `gulp.series()` or `gulp.parallel()` with documentation
* All function definitions for tasks composed of `gulp.series()` or `gulp.parallel()` with documentation
* A `module.exports = {...}` statement at the end without documentation

So something like this?
```js
/**
 * @summary Compile JavaScript files and place them in dist/. Also, generate sourcemaps
 */
const babelCompile = () => gulp.src(['src/**/*.js'])
  .pipe(sourcemaps.init())
  .pipe(gulpBabel())
  .pipe(gulp.dest('dist'));
...

/**
 * @summary Lint and compile, test, and bundle the application
 */
const build = gulp.series(gulp.parallel(lint, typecheck, babel), gulp.parallel(test, bundle));
...

module.exports = {
  babelCompile,
  ...,
  build,
  ...
};
```
What are your thoughts on this?